lib/puppet/indirector/file_content/http.rb

```diff
@@ -10,6 +10,11 @@ class Puppet::Indirector::FileContent::Http < Puppet::Indirector::GenericHttp
 
   @http_method = :get
 
+  def initialize
+    super
+    Puppet.deprecation_warning(_("Puppet::Indirector::FileContent::Http is deprecated. Use Puppet::HTTP::Client instead."))
```

Comment:
should the deprecation warning be before the call to `super`?

Reply:
Why isn't this logged after the terminus class is reset?

Reply:
ah the `Indirection.clear_terminus_class` only clears the current terminus, but doesn't clear the hash of termini that have been created and cached, so it's expected to just see this once.

Reply:
I switched the order. Also there aren't any tests for the http content terminus, and I didn't want to add a file specifically for the deprecation warning.